engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/function/Function.java

```diff
@@ -201,9 +204,28 @@ static Object doCall(
         Function function,
         Object[] arguments,
         @Cached InteropApplicationNode interopApplicationNode,
-        @CachedLibrary("function") InteropLibrary thisLib) {
-      return interopApplicationNode.execute(
-          function, EnsoContext.get(thisLib).emptyState(), arguments);
+        @CachedLibrary("function") InteropLibrary thisLib,
+        @Cached InlinedBranchProfile panicProfile) {
+      try {
+        return interopApplicationNode.execute(
+            function, EnsoContext.get(thisLib).emptyState(), arguments);
+      } catch (StackOverflowError err) {
+        CompilerDirectives.transferToInterpreter();
+        var asserts = false;
+        assert asserts = true;
+        var logger = LoggerFactory.getLogger(Function.class);
+        if (asserts) {
+          logger.error("StackOverflowError detected", err);
+        } else {
+          logger.debug("StackOverflowError detected", err);
+        }
+        throw err;
+      } catch (PanicException ex) {
+        panicProfile.enter(thisLib);
+        // materialize the exception message
+        ex.getMessage();
```

Comment:
Make sure `PanicException.cachedMessage` is computed before leaving the Enso boundary - e.g. most probably before leaving the `org.graalvm.polyglot.Context`.

Reply:
Done in 7c0472b